.evergreen/benchmarks.yml

```diff
@@ -0,0 +1,456 @@
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-standalone"
+    tags: ["5.0", "standalone"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "server"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-replica_set"
+    tags: ["5.0", "replica_set"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "replica_set"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-5.0-sharded_cluster"
+    tags: ["5.0", "sharded_cluster"]
+    commands:
+      - func: "bootstrap mongo-orchestration"
+        vars:
+          MONGODB_VERSION: "5.0"
+          TOPOLOGY: "sharded_cluster"
+      - func: "run driver benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-bson"
+    tags: ["bson"]
+    commands:
+      - func: "run bson benchmarks"
+      - func: "upload benchmark results"
+
+  - name: "benchmark-compile"
+    commands:
+      - func: "compile only"
+      - func: "upload benchmark results"
+
+axes:
+  - id: "mongodb-version"
+    display_name: MongoDB Version
+    values:
+      - id: "5.0"
+        display_name: "5.0"
+        variables:
+          MONGODB_VERSION: "5.0"
+      - id: "4.2"
+        display_name: "4.2"
+        variables:
+          MONGODB_VERSION: "4.2"
+
+  - id: "topology"
+    display_name: Topology
+    values:
+      - id: "standalone"
+        display_name: Standalone
+        variables:
+           TOPOLOGY: "server"
+      - id: "replica-set"
+        display_name: Replica Set
+        variables:
+           TOPOLOGY: "replica_set"
+      - id: "sharded-cluster"
+        display_name: Sharded
+        variables:
+           TOPOLOGY: "sharded_cluster"
+
+  - id: "async-runtime"
+    display_name: Async Runtime
+    values:
+      - id: "tokio"
+        display_name: "tokio"
+        variables:
+          ASYNC_RUNTIME: "tokio"
+      - id: "async-std"
+        display_name: "async-std"
+        variables:
+          ASYNC_RUNTIME: "async-std"
+
+  # Note that drivers-evergreen-tools expects `SSL` as the environmental
+  # variable, not `TLS`, so we have to use that for the actual value used in the
+  # script; we use `TLS` for the metadata that isn't used by the actual shell
+  # scripts.
+  - id: "auth-and-tls"
+    display_name: Authentication and TLS
+    values:
+      - id: "auth-and-tls"
+        display_name: Auth TLS
+        variables:
+           AUTH: "auth"
+           SSL: "ssl"
+
+  - id: "os"
+    display_name: OS
+    values:
+      - id: ubuntu-18.04
+        display_name: "Ubuntu 18.04"
+        run_on: ubuntu1804-test
+        variables:
+          PYTHON: "/opt/mongodbtoolchain/v3/bin/python"
+          VENV_BIN_DIR: "bin"
+      # TODO: RUST-990 re-enable macOS benchmarks
```

Comment:
I opted to just skip macOS benchmarks for now since the timeouts don't seem to be exclusive to the Rust driver (e.g. see the Swift waterfall) and that macOS performance monitoring isn't as high of a priority as the other platforms.